### PR TITLE
CLICKHOUSE-3947 add LIKE and IN () for ODBC and MySQL

### PR DESCRIPTION
**Pull request by VadimPlh**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

**Comment by VadimPE** [comment]
I remember about tests.

File: dbms/src/Storages/transformQueryForExternalDatabase.cpp [hunk 1]
```diff
@@ -81,9 +123,10 @@ String transformQueryForExternalDatabase(
       * copy only compatible parts of it.
       */
 
-    const ASTPtr & original_where = typeid_cast<const ASTSelectQuery &>(query).where_expression;
+    const ASTPtr & original_where = typeid_cast<const ASTSelectQuery &>(*clone_query).where_expression;
```
Comment (alesapin) [inline dbms/src/Storages/transformQueryForExternalDatabase.cpp]: Why we need `const` here If we change that variable inside `replaceConstFunction`?

File: dbms/src/Storages/transformQueryForExternalDatabase.cpp [hunk 2]
```diff
@@ -13,17 +14,57 @@
...
+{
+    for (size_t i = 0; i < node.children.size(); ++i)
+    {
+        auto child = node.children[i];
+        if (const ASTExpressionList * exp_list = typeid_cast<const ASTExpressionList *>(&*child))
+            replaceConstFunction(*exp_list, context, all_columns);
+
+        if (const ASTFunction * function = typeid_cast<const ASTFunction *>(&*child))
+        {
+            auto result_block = getBlockWithConstants(function->clone(), context, all_columns);
```
Comment (alesapin) [inline dbms/src/Storages/transformQueryForExternalDatabase.cpp]: `getBlockWithConstants` doesn't change its first argument. What purpose of `clone`?

**Comment by alesapin** [comment]
lgtm, wait for tests

**Comment by VadimPE** [comment]
> lgtm, wait for tests
https://ru.wikipedia.org/wiki/%D0%96%D0%B4%D1%83%D0%BD#/media/File:Beelden_in_Leiden_2016_04_crop.jpg

**Comment by alexey-milovidov** [comment]
Does not work at all. Tests were bogus and code review was pointless.

**Comment by alexey-milovidov** [comment]
https://github.com/yandex/ClickHouse/pull/5313

**Pull request merged**